cdap-app-fabric/src/main/java/co/cask/cdap/app/mapreduce/MapReduceMetricsInfo.java

```diff
@@ -0,0 +1,217 @@
+    mapContext.put(Constants.Metrics.Tag.MR_TASK_TYPE, MapReduceMetrics.TaskType.Mapper.getId());
+    Map<String, String> reduceContext = Maps.newHashMap(jobContext);
+    reduceContext.put(Constants.Metrics.Tag.MR_TASK_TYPE, MapReduceMetrics.TaskType.Reducer.getId());
+
+    HashMap<String, Long> metrics = Maps.newHashMap();
+
+    metrics.put(TaskCounter.MAP_INPUT_RECORDS.name(),
+                getAggregates(mapContext, MapReduceMetrics.METRIC_INPUT_RECORDS));
+    metrics.put(TaskCounter.MAP_OUTPUT_RECORDS.name(),
+                getAggregates(mapContext, MapReduceMetrics.METRIC_OUTPUT_RECORDS));
+    metrics.put(TaskCounter.MAP_OUTPUT_BYTES.name(),
+                getAggregates(mapContext, MapReduceMetrics.METRIC_BYTES));
+
+    metrics.put(TaskCounter.REDUCE_INPUT_RECORDS.name(),
+                getAggregates(reduceContext, MapReduceMetrics.METRIC_INPUT_RECORDS));
+    metrics.put(TaskCounter.REDUCE_OUTPUT_RECORDS.name(),
+                getAggregates(reduceContext, MapReduceMetrics.METRIC_OUTPUT_RECORDS));
+
+    float mapProgress = getAggregates(mapContext, MapReduceMetrics.METRIC_COMPLETION) / 100.0F;
+    float reduceProgress = getAggregates(reduceContext, MapReduceMetrics.METRIC_COMPLETION) / 100.0F;
+
+    return new MRJobInfo(null, null, null, mapProgress, reduceProgress, metrics, mapTaskInfos, reduceTaskInfos);
+  }
+
+  private String prependSystem(String metric) {
+    return "system." + metric;
+  }
+
+  private long getAggregates(Map<String, String> context, String metric) throws Exception {
+    MetricDataQuery metricDataQuery =
+      new MetricDataQuery(0, Integer.MAX_VALUE, Integer.MAX_VALUE, prependSystem(metric), MetricType.COUNTER, context,
+                          ImmutableList.<String>of());
+    Collection<MetricTimeSeries> query = metricStore.query(metricDataQuery);
+    if (query.isEmpty()) {
+      return 0;
+    }
+    MetricTimeSeries timeSeries = Iterables.getOnlyElement(query);
+    List<TimeValue> timeValues = timeSeries.getTimeValues();
+    TimeValue timeValue = Iterables.getOnlyElement(timeValues);
+    return timeValue.getValue();
+  }
+
+  // queries MetricStore for one metric across all tasks of a certain TaskType, using GroupBy InstanceId
+  private Map<String, Long> queryGroupedAggregates(Map<String, String> context, String metric) throws Exception {
+    MetricDataQuery metricDataQuery =
+      new MetricDataQuery(0, Integer.MAX_VALUE, Integer.MAX_VALUE, prependSystem(metric), MetricType.GAUGE, context,
+                          ImmutableList.of(Constants.Metrics.Tag.INSTANCE_ID));
+    Collection<MetricTimeSeries> query = metricStore.query(metricDataQuery);
+
+    // runId -> metricValue
+    Map<String, Long> taskMetrics = Maps.newHashMap();
+    for (MetricTimeSeries metricTimeSeries : query) {
+      List<TimeValue> timeValues = metricTimeSeries.getTimeValues();
+      TimeValue timeValue = Iterables.getOnlyElement(timeValues);
```

Comment:
This should always be true right? (unless there's a problem with the MetricStore, because I'm asking for aggregate).
So probably throw an exception if it isn't true.
What do you think?